src/dash-table/derived/header/content.tsx

```diff
@@ -79,14 +104,22 @@ function getSortingIcon(columnId: ColumnId, sortBy: SortBy) {
     }
 }
 
+function getColumnFlag(i: number, flag?: boolean | boolean[]): boolean {
+    return typeof flag === 'boolean' ?
+        flag :
+        !!flag && flag[i];
+}
```

Comment:
renamable, clearable, deletable all use the same logic to decide whether the action is available